app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/usos/data/api/UsosApiUser.kt

```diff
@@ -53,7 +53,12 @@ class UsosApiUser(
             data.studentId = json.getInt("id") ?: data.studentId
             profile?.studentNameLong = studentName
             profile?.studentNameShort = studentName.getShortName()
-            profile?.studentNumber = json.getInt("student_number", -1)
+            val studentNumWithoutNonDigits = json.getString("student_number")?.replace(Regex("[^0-9]"), "")
+            if (studentNumWithoutNonDigits != null) {
```

Comment:
It would be better to check if the number is null OR empty/blank - if there are only letters in the number, we don't want to convert empty string to int.

Reply:
I will fix it